### PR TITLE
OpenShift native signatures

### PR DESCRIPTION
**Pull request by mtrmac**
This adds support for storing signatures in Atomic Registry, assuming server-side support per (merged) openshift/origin#8371 and (currently unmerged) openshift/origin#9181 .

(Marked as WIP because the server side is missing, and because this currently requires a cluster-wide “update signatures” privilege, which seems impractical, so future changes of the API cannot be ruled out. Also, this is currently done as 2 commits to preserve the 2 possible ways to upload signatures, after the decisions are final they should be squashed.)

### CODE REVIEW

**Comment by philips** [comment]
cc @dgonyeo @ecordell @gtank

File: openshift/openshift.go [hunk 1]
```diff
@@ -365,6 +434,20 @@ type image struct {
...
+
+type imageSignature struct {
+	Type    string `json:"type"`
+	Content []byte `json:"content"`
+	// Conditions []SignatureCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// ImageIdentity string `json:"imageIdentity,omitempty"`
+	// SignedClaims map[string]string `json:"signedClaims,omitempty"`
+	// Created *unversioned.Time `json:"created,omitempty"`
+	// IssuedBy SignatureIssuer `json:"issuedBy,omitempty"`
+	// IssuedTo SignatureSubject `json:"issuedTo,omitempty"`
```
Comment (dgonyeo) [inline openshift/openshift.go]: What's up with all of these commented out fields? Something to be used in the future?

Comment (mtrmac) [inline openshift/openshift.go]: Those fields exist in the upstream OpenShift objects, and we do not use them (In particular these should be filled in by the server, to have something to display in an UI, so we want to neither fill them in (the server will override that) nor read them (we want to make independent policy decisions without trusting the server)).

This ~ only acknowledges their existence, and unlike uncommented fields we do not need to define the used types etc.

**Comment by mtrmac** [comment]
Updated against the final merged API from https://github.com/openshift/origin/pull/9181 .

I'm afraid this still requires a cluster-wide signature privilege, but it does work and it is hopefully already useful.

**Comment by mtrmac** [comment]
See also https://github.com/projectatomic/skopeo/pull/93 for the necessary `copy.go` update.  (Because we don’t yet have tests of signature handling, not merging that doesn’t show up as a test failure. Tests exist in https://github.com/mtrmac/skopeo/tree/integrate-all-the-things , and are ~blocked on really finishing https://github.com/mtrmac/skopeo/tree/verify-on-pull ).

**Comment by runcom** [comment]
lgtm

**Comment by mtrmac** [comment]
> Because we don’t yet have tests of signature handling, not merging that doesn’t show up as a test failure.

Actually it does, because even `PutSignatures(empty_slice)` triggers the ordering check. Anyway, https://github.com/projectatomic/skopeo/pull/93 fixes that.

**Comment by mtrmac** [comment]
:+1:

**Pull request merged**